[harfbuzz] Build error

Copy issue body from C:/vcpkg/installed/vcpkg/issue_body.md

-- Package x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: C:/vcpkg/downloads/tools/ninja/1.10.2-windows/ninja.exe install -v
    Working Directory: C:/vcpkg/buildtrees/harfbuzz/x64-windows-dbg
    Error code: 1
    See logs for more information:
      C:\vcpkg\buildtrees\harfbuzz\package-x64-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_meson.cmake:33 (vcpkg_execute_required_process)
  ports/harfbuzz/portfile.cmake:70 (vcpkg_install_meson)
  scripts/ports.cmake:147 (include)


error: building harfbuzz:x64-windows failed with: BUILD_FAILED
Elapsed time to handle harfbuzz:x64-windows: 2.9 min


Here the compile error out of log file:
        with
        [
            Type=void
        ]
C:\vcpkg\buildtrees\harfbuzz\src\8.2.1-63f9ae7085.clean\src\hb-subset.cc : fatal error C1128: number of sections exceeded object file format limit: compile with /bigobj
ninja: build stopped: subcommand failed.

--- a/ports/harfbuzz/fix-win32-build.patch
+++ b/ports/harfbuzz/fix-win32-build.patch
@@ -1,0 +1,12 @@
+diff --git a/src/meson.build b/src/meson.build
+index 4d63ecf..266c213 100644
+--- a/src/meson.build
++++ b/src/meson.build
+@@ -561,6 +561,7 @@ version = '0.@0@.0'.format(hb_version_int)
+ 
+ extra_hb_cpp_args = []
+ if cpp.get_argument_syntax() == 'msvc'
++  extra_hb_cpp_args += ['/bigobj']
+   if get_option('default_library') != 'static'
+     extra_hb_cpp_args += '-DHB_DLL_EXPORT'
+   endif

--- a/ports/harfbuzz/fix-win32-build.patch
+++ b/ports/harfbuzz/fix-win32-build.patch
@@ -1,12 +1,26 @@
 diff --git a/src/meson.build b/src/meson.build
-index 4d63ecf..266c213 100644
+index 87e8962..bdfa797 100644
 --- a/src/meson.build
 +++ b/src/meson.build
-@@ -561,6 +561,7 @@ version = '0.@0@.0'.format(hb_version_int)
+@@ -387,6 +387,11 @@ hb_subset_sources = files(
+   'hb-subset.hh',
+ )
  
- extra_hb_cpp_args = []
- if cpp.get_argument_syntax() == 'msvc'
++extra_hb_cpp_args = []
++if cpp.get_argument_syntax() == 'msvc'
 +  extra_hb_cpp_args += ['/bigobj']
++endif
++
+ hb_subset_headers = files(
+   'hb-subset.h',
+   'hb-subset-repacker.h'
+@@ -559,8 +564,7 @@ defs_list = [harfbuzz_def]
+ 
+ version = '0.@0@.0'.format(hb_version_int)
+ 
+-extra_hb_cpp_args = []
+-if cpp.get_define('_MSC_FULL_VER') != ''
++if cpp.get_argument_syntax() == 'msvc'
    if get_option('default_library') != 'static'
      extra_hb_cpp_args += '-DHB_DLL_EXPORT'
    endif

--- a/ports/harfbuzz/portfile.cmake
+++ b/ports/harfbuzz/portfile.cmake
@@ -4,6 +4,8 @@ vcpkg_from_github(
     REF ${VERSION}
     SHA512 23d6abbd270885d7ae1ebb3c981f0c331a48d891e23caffe9e254f5e7e205bb0348add7b371526166a49b336f8076f92c11ef76ca81f48a6fd9f58812ec96d79
     HEAD_REF master
+    PATCHES
+        fix-win32-build.patch
 )
 
 if("icu" IN_LIST FEATURES)

--- a/ports/harfbuzz/vcpkg.json
+++ b/ports/harfbuzz/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "harfbuzz",
   "version": "8.2.1",
+  "port-version": 1,
   "description": "HarfBuzz OpenType text shaping engine",
   "homepage": "https://github.com/harfbuzz/harfbuzz",
   "license": "MIT-Modern-Variant",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3186,7 +3186,7 @@
     },
     "harfbuzz": {
       "baseline": "8.2.1",
-      "port-version": 0
+      "port-version": 1
     },
     "hash-library": {
       "baseline": "8",

--- a/versions/h-/harfbuzz.json
+++ b/versions/h-/harfbuzz.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "b5c3272f29b8d4a9d24e475c13982825cec6377a",
+      "version": "8.2.1",
+      "port-version": 1
+    },
+    {
       "git-tree": "6894af0b171aea403718ebb131b454a60b0c961a",
       "version": "8.2.1",
       "port-version": 0

--- a/versions/h-/harfbuzz.json
+++ b/versions/h-/harfbuzz.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "b5c3272f29b8d4a9d24e475c13982825cec6377a",
+      "git-tree": "1f39a3081e7055cbf5e8cc27b5f91fef2a274415",
       "version": "8.2.1",
       "port-version": 1
     },

